src/mongo/delegates/Views.class.php

```diff
@@ -396,58 +406,16 @@ public function generateView($viewId,$resource=null,$context=null,$queueName=nul
             $t = new \Tripod\Timer();
             $t->start();
 
-            $from = $this->getFromCollectionForViewSpec($viewSpec);
+            $from = $viewSpec['from'];
             $collection = $this->config->getCollectionForView($this->storeName, $viewId);
 
+            $this->ensureIndices($viewSpec, $collection);
```

Comment:
Ok.  I can see that there's not an "on deployment, run this" type of job where we could naturally hook that in, so I accept this approach is sensible.  Thanks for clarifying!

Would you object to me centralising the shared indices to reduce code, as I have done with `Config::ensureIndexForSpecRevision`?